packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt

```diff
@@ -42,8 +42,8 @@ object DatadogSDKWrapperStorage {
      * Exposed for testing purposes only.
      */
     fun notifyOnInitializedListeners(ddCore: InternalSdkCore) {
-        onInitializedListeners.forEach {
```

Comment:
what is the reason for this change? this `forEach` is coming from the Kotlin stdlib (if I'm not wrong) and shouldn't be impacted by the Android stdlib API availability (same for other similar changes).

Reply:
When running `cd packages/core/android &&./gradlew :lintDebug` we would get the following errors for `forEach` calls:

```
/Users/louis.zawadzki/go/src/github.com/DataDog/mobile/dd-sdk-reactnative/packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt:45: Error: Call requires API level 24 (current min is 21): java.lang.Iterable#forEach [NewApi]
        onInitializedListeners.forEach {
                               ~~~~~~~
```

Reply:
this is pretty weird why here the method from JDK is used, because we use `forEach` pretty much a lot in the native SDK where min API is 21 and there are no problems.

The method from Kotlin stdlib is like that:

```
@kotlin.internal.HidesMembers
public inline fun <T> Iterable<T>.forEach(action: (T) -> Unit): Unit {
    for (element in this) action(element)
}
```

I'm pretty sure root cause is somewhere else.